node/src/components/chain_synchronizer/operations.rs

```diff
@@ -850,17 +851,35 @@ where
                     );
                 }
 
-                if let Err(error) = consensus::check_sufficient_finality_signatures(
+                if item.block_signatures().proofs.is_empty() {
+                    warn!(?peer, ?item, "no block signatures from peer");
+                    ctx.effect_builder.announce_disconnect_from_peer(peer).await;
+                    continue;
+                }
+
+                match consensus::check_sufficient_finality_signatures(
```

Comment:
```suggestion
                match consensus::check_sufficient_block_signatures(
```

And we could move this out of `consensus`, too. (Maybe to `linear_chain`?)

Reply:
Agreed, and done.